tiledb/sm/filesystem/vfs.cc

```diff
@@ -802,6 +803,10 @@ Status VFS::init(
   posix_.init(config_, io_tp_);
 #endif
 
+  if (memfs_root != nullptr) {
+    memfs_.init(memfs_root);
```

Comment:
So far `memfs` was C41 compliant afaiu, and it's a pity that we introduce a 2 step process with the `init` in this fix.
Can't we instead change `VFS::VFS()` constructor to take `memfs_root`, that defaults to `nullptr` as well, and then passes it to the Constructor of `MemFilesystem::MemFilesystem()` that we'll adapt to take an argument also etc. ?

Reply:
I initially was working on an implementation with a memfs constructor that takes in a root instead of default initializing it. I couldn't get that implementation to work (which is why this work item took so long) but I'll hack at it again

Reply:
Thanks, let me know if I can help!

Reply:
Implementation working on shared_ptr, trying to get same results for unique_ptr